src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -634,6 +591,75 @@ def get_objects_owned_by_application(self) -> List[ApplicationOwnedObject]:
             ).fetchall()
             return [{"name": row[1], "type": row[2]} for row in results]
 
+    def _upgrade_app(
+        self,
+        stage_fqn: str,
+        install_method: SameAccountInstallMethod,
+        event_sharing: EventSharingHandler,
+        current_app_row: dict,
+        policy: PolicyBase,
+        interactive: bool,
+    ) -> list[tuple[str]] | None:
+        console = self._workspace_ctx.console
+        self._workspace_ctx.console.step(
+            f"Upgrading existing application object {self.name}."
+        )
+
+        try:
+            return get_snowflake_facade().upgrade_application(
+                name=self.name,
+                current_app_row=current_app_row,
+                install_method=install_method,
+                stage_fqn=stage_fqn,
+                debug_mode=self._entity_model.debug,
```

Comment:
kinda nit: 
create a `@property` getter for `self._entity_model.debug`, same for `self._entity_model.telemetry`

`self._workspace_ctx.console` is used in various places too.